File: src/test/java/org/oewntk/model/LibTestModelSenseGroups.java
/*
 * Copyright (c) 2021. Bernard Bou.
 */

package org.oewntk.model;

import java.io.PrintStream;

import static org.oewntk.model.SenseGroupings.sensesForLCLemma;
import static org.oewntk.model.SenseGroupings.sensesForLCLemmaAndPos;

public class LibTestModelSenseGroups
{
	public static void testCISensesGroupingByLCLemmaAndPos(final CoreModel model, final String word, final char pos, final PrintStream ps)
	{
		ps.printf("ci '%s' %s%n", word, pos);
		ps.println(testCISensesGroupingByLCLemmaAndPosString(model, word, pos));
	}

	public static void testCISensesGroupingByLCLemma(final CoreModel model, final String word, final PrintStream ps)
	{
		ps.printf("ci '%s'%n", word);
		ps.println(testCISensesGroupingByLCLemmaString(model, word));
	}

	private static String testCISensesGroupingByLCLemmaString(final CoreModel model, final String word)
	{
		final var senses = sensesForLCLemma(model.senses, word);
		return Utils.sensesToStringByDecreasingTagCount(senses);
	}

	private static String testCISensesGroupingByLCLemmaAndPosString(final CoreModel model, final String word, final char pos)
	{
		final var senses = sensesForLCLemmaAndPos(model.senses, word, pos);
		return Utils.sensesToStringByDecreasingTagCount(senses);
	}
}

File: src/test/java/org/oewntk/model/LibTestModelSenseGroups.kt
/*
 * Copyright (c) 2024-2024. Bernard Bou.
 */

package org.oewntk.model

import org.oewntk.model.SenseGroupings.sensesForLCLemma
import org.oewntk.model.SenseGroupings.sensesForLCLemmaAndPos
import org.oewntk.model.TestUtils.sensesToStringByDecreasingTagCount
import java.io.PrintStream

object LibTestModelSenseGroups {

    fun testCISensesGroupingByLCLemmaAndPos(model: CoreModel, word: Lemma, category: Category, ps: PrintStream) {
        ps.printf("ci '%s' %s%n", word, category)
        ps.println(testCISensesGroupingByLCLemmaAndPosString(model, word, category))
    }

    fun testCISensesGroupingByLCLemma(model: CoreModel, word: Lemma, ps: PrintStream) {
        ps.printf("ci '%s'%n", word)
        ps.println(testCISensesGroupingByLCLemmaString(model, word))
    }

    private fun testCISensesGroupingByLCLemmaString(model: CoreModel, lemma: Lemma): String {
        val senses = sensesForLCLemma(model.senses, lemma)
        return sensesToStringByDecreasingTagCount(senses)
    }

    private fun testCISensesGroupingByLCLemmaAndPosString(model: CoreModel, lemma: Lemma, category: Category): String {
        val senses = sensesForLCLemmaAndPos(model.senses, lemma, category)
        return sensesToStringByDecreasingTagCount(senses)
    }
}
